modules/homa-lite/src/lib.rs

```diff
@@ -0,0 +1,253 @@
+
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+//! # Homa Lite Module
+//! The Homa Lite module handles logic that allows the users to lock in KSM tokens on the Karura
+//! Acala Chain, and mint LKSM tokens from the liquidity. The locked KSM are then used for Staking -
+//! they will be used to nominate our partner Validators on the Kusama Chain.
+//!
+//! As the first draft, this module currently does not support Redeem function from LKSM to KSM.
+//!
+//! General workflow:
+//! 1. User moves KSM cross-chain into the Karura chain
+//! 2. User "Lock" their KSM on the Karura chain
+//! 3. Karura send XCM back into Kusama chain, and Nominate these KSMs against our partner
+//! Validators. 4. Karura mint LKSM on the Karura chain
+
+#![cfg_attr(not(feature = "std"), no_std)]
+#![allow(clippy::unused_unit)]
+
+mod mock;
+mod tests;
+
+use frame_support::{pallet_prelude::*, transactional, PalletId};
+use frame_system::{ensure_signed, pallet_prelude::*};
+use orml_traits::MultiCurrency;
+use primitives::{Balance, CurrencyId, EraIndex};
+use sp_std::prelude::*;
+
+pub use module::*;
+
+#[frame_support::pallet]
+pub mod module {
+	use super::*;
+
+	#[derive(Encode, Decode, Clone, RuntimeDebug, PartialEq, Eq)]
+	pub struct TotalIssuanceInfo {
+		pub staking_total: Balance,
+		pub liquid_total: Balance,
+	}
+
+	#[pallet::config]
+	pub trait Config: frame_system::Config {
+		type Event: From<Event<Self>> + IsType<<Self as frame_system::Config>::Event>;
+
+		/// Multi-currency support for asset management
+		type Currency: MultiCurrency<Self::AccountId, CurrencyId = CurrencyId, Balance = Balance>;
+
+		/// The Currency ID for the Staking asset
+		#[pallet::constant]
+		type StakingCurrencyId: Get<CurrencyId>;
+
+		/// The Currency ID for the Liquid asset
+		#[pallet::constant]
+		type LiquidCurrencyId: Get<CurrencyId>;
+
+		/// The ID for this pallet
+		#[pallet::constant]
+		type PalletId: Get<PalletId>;
+
+		/// Origin used to Issue LKSM
+		type IssuerOrigin: EnsureOrigin<Self::Origin>;
+
+		/// Origin represented by the Root or Governance
+		type GovernanceOrigin: EnsureOrigin<Self::Origin>;
+	}
+
+	#[pallet::error]
+	pub enum Error<T> {
+		/// The current Era has not finished, therefore the Liquid currency have not been issued
+		/// yet.
+		LiquidCurrencyNotIssuedForThisEra,
+		/// Mathematical error occurred during calculation.
+		ArithmeticError,
```

Comment:
no need to have this

https://github.com/paritytech/substrate/blob/77dcc4f90917f2215ee40efeacd68be9ce85db14/primitives/runtime/src/lib.rs#L571